src/pages/index.js

```diff
@@ -0,0 +1,323 @@
+    handleImageClick(data);
+  });
+
+  return cardElement;
+}
+
+function handleCardLike(evt, dataId) {
+  let isLiked = evt.target.classList.contains("card__like-button_liked")
+    ? true
+    : false;
+  api
+    .toggleLike(dataId, isLiked)
+    .then(() => evt.target.classList.toggle("card__like-button_liked"))
+    .catch(console.error);
+}
+
+function handleImageClick(data) {
+  openModal(previewModal);
+  previewModalImageEl.src = data.link;
+  previewModalCaptionEl.textContent = data.name;
+  previewModalImageEl.alt = data.name;
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Deleting...";
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally((submitButton.textContent = "Delete"));
+}
+
+deleteForm.addEventListener("submit", handleDeleteSubmit);
+
+function handleOutsideClick(modal) {
+  return function (event) {
+    if (!event.target.closest(".modal__content")) {
+      closeModal(modal);
+    }
+  };
+}
+
+function handleEscapePress(modal) {
+  return function (event) {
+    if (event.key === "Escape" || event.key === "Esc") {
+      closeModal(modal);
+    }
+  };
+}
+
+let outsideClickHandler, escapePressHandler;
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+
+  outsideClickHandler = handleOutsideClick(modal);
+  escapePressHandler = handleEscapePress(modal);
+
+  setTimeout(() => {
+    addListeners(outsideClickHandler, escapePressHandler);
+  }, 1);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+
+  if (outsideClickHandler && escapePressHandler) {
+    removeListeners(outsideClickHandler, escapePressHandler);
+  }
+
+  outsideClickHandler = null;
+  escapePressHandler = null;
+}
+
+function addListeners(outsideClickHandler, escapePressHandler) {
+  document.addEventListener("click", outsideClickHandler);
+  document.addEventListener("keydown", escapePressHandler);
+}
+
+function removeListeners(outsideClickHandler, escapePressHandler) {
+  document.removeEventListener("click", outsideClickHandler);
+  document.removeEventListener("keydown", escapePressHandler);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      disableButton(profileSubmitButton, settings);
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally((submitButton.textContent = "Save"));
```

Comment:
In `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 


```js
    .finally(() => {
      submitButton.textContent = "Save";
    });

```


**Please, correct it everywhere in the code**

Reply:
fixed